2022-12-29 17:48:01.658 | ERROR    | sflow.except_hook_handler:__handle_exception:21 - type:<class 'TypeError'>; value:new_env() got an unexpected keyword argument 'wl_start'; traceback:['  File "C:\\photoCAD\\layout1002\\sim_circuit_demux.py", line 129, in <module>\n    fp.export_spc(device, file=spc_filename, components=components, sim_env=fp.sim.Env(**env))\n']
2022-12-29 17:56:47.623 | ERROR    | sflow.except_hook_handler:__handle_exception:21 - type:<class 'AttributeError'>; value:'NoneType' object has no attribute 'ports'; traceback:['  File "C:\\photoCAD\\layout1002\\sim_circuit_demux.py", line 131, in <module>\n    sim_result = sf.run_sim(\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\run_sim.py", line 23, in run_sim\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\run_sim.py", line 52, in __parse_for_sim\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\simulation\\psim_parser.py", line 66, in parse\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\parser.py", line 74, in parse\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\parser.py", line 566, in _find_expose_ports\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\parser.py", line 590, in _find_expose_ports_by_ele\n']
2023-01-11 17:06:07.892 | ERROR    | sflow.except_hook_handler:__handle_exception:21 - type:<class 'TypeError'>; value:new_env() missing 3 required keyword-only arguments: 'wl_start', 'wl_end', and 'points_num'; traceback:['  File "C:\\photoCAD\\layout1002\\sim_circuit_demux4.py", line 207, in <module>\n    fp.export_spc(device, file=spc_filename, components=components, sim_env=fp.sim.Env(wavelength=env.wl_range))\n']
2023-02-22 14:31:26.778 | ERROR    | sflow.except_hook_handler:__handle_exception:21 - type:<class 'NameError'>; value:name 'Path' is not defined; traceback:['  File "c:/photoCAD/layout1002/.venv_layout1002/Lib/site-packages/gpdk/examples/mzisimulation.py", line 71, in <module>\n    fp.export_spc(device, file=spc_filename, components=components, sim_env=fp.sim.Env(**env))\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\fnpcell\\internal\\util\\spc_util.py", line 58, in export_spc\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\fnpcell\\interfaces.py", line 303, in run\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\fnpcell\\netlist\\spc_writer.py", line 172, in enter\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\fnpcell\\netlist\\spc_writer.py", line 99, in write_subcircuit\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\gpdk\\components\\mmi\\mmi.py", line 91, in sim_model\n    file_path = Path("Mmi1x2").with_suffix(".dat")\n']
2023-02-22 14:35:38.365 | ERROR    | sflow.except_hook_handler:__handle_exception:21 - type:<class 'NameError'>; value:name 'Path' is not defined; traceback:['  File "c:/photoCAD/layout1002/.venv_layout1002/Lib/site-packages/gpdk/examples/mzisimulation.py", line 71, in <module>\n    fp.export_spc(device, file=spc_filename, components=components, sim_env=fp.sim.Env(**env))\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\fnpcell\\internal\\util\\spc_util.py", line 58, in export_spc\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\fnpcell\\interfaces.py", line 303, in run\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\fnpcell\\netlist\\spc_writer.py", line 172, in enter\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\fnpcell\\netlist\\spc_writer.py", line 99, in write_subcircuit\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\gpdk\\components\\mmi\\mmi.py", line 91, in sim_model\n    file_path = Path("Mmi1x2").with_suffix(".dat")\n']
2023-04-18 11:25:43.124 | ERROR    | sflow.except_hook_handler:__handle_exception:21 - type:<class 'NameError'>; value:name 'Path' is not defined; traceback:['  File "C:\\photoCAD\\layout1002\\sim_circuit_demux.py", line 135, in <module>\n    fp.export_spc(device, file=spc_filename, components=components, sim_env=fp.sim.Env(**env))\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\fnpcell\\internal\\util\\spc_util.py", line 59, in export_spc\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\fnpcell\\interfaces.py", line 303, in run\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\fnpcell\\netlist\\spc_writer.py", line 172, in enter\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\fnpcell\\netlist\\spc_writer.py", line 99, in write_subcircuit\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\gpdk\\components\\directional_coupler\\directional_coupler_bend.py", line 128, in sim_model\n    file_path = Path("DC_050").with_suffix(".dat")\n']
2023-04-18 11:29:48.728 | ERROR    | sflow.except_hook_handler:__handle_exception:21 - type:<class 'TypeError'>; value:new_env() missing 3 required keyword-only arguments: 'wl_start', 'wl_end', and 'points_num'; traceback:['  File "C:\\photoCAD\\layout1002\\sim_circuit_demux4.py", line 208, in <module>\n    fp.export_spc(device, file=spc_filename, components=components, sim_env=fp.sim.Env(wavelength=env.wl_range))\n']
2023-04-18 11:33:35.123 | ERROR    | sflow.except_hook_handler:__handle_exception:21 - type:<class 'TypeError'>; value:compiled_function object argument after ** must be a mapping, not PsimEnvironment; traceback:['  File "C:\\photoCAD\\layout1002\\sim_circuit_demux4.py", line 208, in <module>\n    fp.export_spc(device, file=spc_filename, components=components, sim_env=fp.sim.Env(**env))\n']
2023-04-18 11:35:32.575 | ERROR    | sflow.except_hook_handler:__handle_exception:21 - type:<class 'IndexError'>; value:list index out of range; traceback:['  File "C:\\photoCAD\\layout1002\\sim_circuit_demux4.py", line 216, in <module>\n    sim_result = sf.run_sim(\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\run_sim.py", line 23, in run_sim\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\run_sim.py", line 54, in __parse_for_sim\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\simulation\\psim_parser.py", line 120, in to_python\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\simulation\\psim_parser.py", line 267, in _to_python_source_component\n']
2023-04-18 11:39:53.193 | ERROR    | sflow.except_hook_handler:__handle_exception:21 - type:<class 'IndexError'>; value:list index out of range; traceback:['  File "C:\\photoCAD\\layout1002\\sim_circuit_demux4.py", line 216, in <module>\n    sim_result = sf.run_sim(\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\run_sim.py", line 23, in run_sim\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\run_sim.py", line 54, in __parse_for_sim\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\simulation\\psim_parser.py", line 120, in to_python\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\simulation\\psim_parser.py", line 267, in _to_python_source_component\n']
2023-04-18 11:46:16.247 | ERROR    | sflow.except_hook_handler:__handle_exception:21 - type:<class 'IndexError'>; value:list index out of range; traceback:['  File "C:\\photoCAD\\layout1002\\sim_circuit_demux4.py", line 217, in <module>\n    sim_result = sf.run_sim(\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\run_sim.py", line 23, in run_sim\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\run_sim.py", line 54, in __parse_for_sim\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\simulation\\psim_parser.py", line 120, in to_python\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\simulation\\psim_parser.py", line 267, in _to_python_source_component\n']
2023-04-18 11:50:36.972 | ERROR    | sflow.except_hook_handler:__handle_exception:21 - type:<class 'TypeError'>; value:compiled_function object argument after ** must be a mapping, not PsimEnvironment; traceback:['  File "C:\\photoCAD\\layout1002\\sim_circuit_demux4.py", line 209, in <module>\n    fp.export_spc(device, file=spc_filename, components=components, sim_env=fp.sim.Env(**env))\n']
2023-04-18 11:53:18.566 | ERROR    | sflow.except_hook_handler:__handle_exception:21 - type:<class 'TypeError'>; value:new_env() missing 3 required keyword-only arguments: 'wl_start', 'wl_end', and 'points_num'; traceback:['  File "C:\\photoCAD\\layout1002\\sim_circuit_demux4.py", line 208, in <module>\n    fp.export_spc(device, file=spc_filename, components=components, sim_env=fp.sim.Env(wavelength=env.wl_range))\n']
2023-04-18 13:35:30.906 | ERROR    | sflow.except_hook_handler:__handle_exception:21 - type:<class 'TypeError'>; value:new_env() missing 3 required keyword-only arguments: 'wl_start', 'wl_end', and 'points_num'; traceback:['  File "C:\\photoCAD\\layout1002\\sim_circuit_demux4.py", line 208, in <module>\n    fp.export_spc(device, file=spc_filename, components=components, sim_env=fp.sim.Env(wavelength=env.wl_range))\n']
2023-04-18 13:36:58.338 | ERROR    | sflow.except_hook_handler:__handle_exception:21 - type:<class 'IndexError'>; value:list index out of range; traceback:['  File "C:\\photoCAD\\layout1002\\sim_circuit_demux4.py", line 218, in <module>\n    sim_result = sf.run_sim(\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\run_sim.py", line 23, in run_sim\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\run_sim.py", line 54, in __parse_for_sim\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\simulation\\psim_parser.py", line 120, in to_python\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\simulation\\psim_parser.py", line 267, in _to_python_source_component\n']
2023-04-18 13:41:14.273 | ERROR    | sflow.except_hook_handler:__handle_exception:21 - type:<class 'IndexError'>; value:list index out of range; traceback:['  File "C:\\photoCAD\\layout1002\\sim_circuit_demux4.py", line 220, in <module>\n    sim_result = sf.run_sim(\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\run_sim.py", line 23, in run_sim\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\run_sim.py", line 54, in __parse_for_sim\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\simulation\\psim_parser.py", line 120, in to_python\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\simulation\\psim_parser.py", line 267, in _to_python_source_component\n']
2023-04-24 11:50:40.713 | ERROR    | sflow.except_hook_handler:__handle_exception:21 - type:<class 'UnboundLocalError'>; value:local variable 'file_path' referenced before assignment; traceback:['  File "C:\\photoCAD\\layout1002\\sim_circuit_demux.py", line 135, in <module>\n    fp.export_spc(device, file=spc_filename, components=components, sim_env=fp.sim.Env(**env))\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\fnpcell\\internal\\util\\spc_util.py", line 59, in export_spc\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\fnpcell\\interfaces.py", line 303, in run\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\fnpcell\\netlist\\spc_writer.py", line 172, in enter\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\fnpcell\\netlist\\spc_writer.py", line 99, in write_subcircuit\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\gpdk\\components\\directional_coupler\\directional_coupler_bend.py", line 113, in sim_model\n    return fp.sim.ExternalFileModel(file_path)\n']
2023-05-12 21:21:27.571 | ERROR    | sflow.except_hook_handler:__handle_exception:21 - type:<class 'TypeError'>; value:run_sim() got an unexpected keyword argument 'is_print_netlist'; traceback:['  File "C:\\photoCAD\\layout1002\\sim_circuit_demux.py", line 137, in <module>\n    sim_result = sf.run_sim(\n']
2023-05-12 21:21:41.917 | ERROR    | psim.except_hook_handler:__handle_exception:22 - type:<class 'ValueError'>; value:U1049 DirectionalCouplerBend sim_model=None is not supported!; traceback:['  File "C:\\photoCAD\\layout1002\\sim_circuit_demux.py", line 137, in <module>\n    sim_result = sf.run_sim(\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\run_sim.py", line 23, in run_sim\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\run_sim.py", line 52, in __parse_for_sim\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\simulation\\psim_parser.py", line 68, in parse\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\parser.py", line 79, in parse\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\parser.py", line 371, in _annotate_element\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\parser.py", line 395, in _annotate_element_act\n', '  File "C:\\photoCAD\\layout1002\\.venv_layout1002\\lib\\site-packages\\sflow\\parser.py", line 425, in _set_component_for_element\n']
